cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java

```diff
@@ -180,9 +218,24 @@ public void enforce(EntityId entityId, Principal principal, Set<Action> actions)
     Preconditions.checkArgument(Principal.PrincipalType.USER == principal.getType(), "The given principal '%s' is of " +
                                 "type '%s'. In Sentry authorization checks can only be performed on principal type " +
                                 "'%s'.", principal.getName(), principal.getType(), Principal.PrincipalType.USER);
-    if (!binding.authorize(entityId, principal, actions)) {
-      throw new UnauthorizedException(principal, actions, entityId);
+    Set<Action> disallowed = EnumSet.noneOf(Action.class);
```

Comment:
How does this cache works ? On enforce its loaded with the AuthorizationPrivilege and if the authorization is not  there the boolean says false. Now when you do grant don't you need to refresh/invalidate the cache else the cache will stay say unauthorized ?